deliverables/best_practices.md

```diff
@@ -125,3 +125,106 @@ Always choose the version of Python that is correct for your usage.
   The `typing` module was new in Python 3.5
   but additional features from the Python 3.8 version are also backported.
   See https://docs.python.org/3.8/library/typing.html.
+
+## Avoid use of futurize stage 2 and its standard library backports
+
+[futurize](https://python-future.org/overview.html#automatic-conversion-to-py2-3-compatible-code)
+can be helpful for transitioning standalone projects and scripts to
+Python 3, but `--futurize --stage2` adds dependencies on the
+`future.standard_library` module that changes the behavior of the Python
+standard library.  This makes it unsuitable for use in libraries or plugins
+that have to co-exist alongside other python plugins in a host application.
+
+It is preferrable to use `six` and the standard library instead of the
+`future`, `past`, and `builtins` modules provided by `python-future`.
+
+## Read the Conservative Python 3 Porting Guide
+
+[The Conservative Python 3 Porting Guide](https://portingguide.readthedocs.io/en/latest/)
+(CPPG) is an invaluable resource for porting to Python 3.
+This document borrows heavily from the CPPG recommendations.
+
+## Read the futurize Python 3 Compatibility Cheat Sheet
+
+The futurize project contains a
+[Cheat Sheet](https://python-future.org/compatible_idioms.html)
+for writing Python 2 + 3 compatible code that contains a lot of helpful
+examples for transitioning Python 2-only code to be 2 + 3 compatible.
+
+Examples using `builtins`, `past` and `future` should be avoided.
+Equivalent behavior can be attained using `six` and builtin types.
+
+## Use pylint's py3k mode to catch compatibility issues
+
+Pylint contains a "py3k" mode that enables Python 2 + 3 compatibility checks
+and is extremely helpful when transitioning to Python 2 + 3 compatibility.
+Use of `pylint --py3k` is strongly encouraged.
```

Comment:
It's also useful as a way of ensuring that Python 3 compatibility remains in a repository, especially when there aren't many tests. Is this worth mentioning?